### PR TITLE
Akka.Remote.TestKit to Protobuf3

### PR DESCRIPTION
**Pull request by heynickc**
Adds to work previously done by @alexvaluyskiy to upgrade Akka.Remote.TestKit to Protobuf3 to enable MultiNode Test Runner to target .NET Core.

### CODE REVIEW

**Comment by alexvaluyskiy** [comment]
@heynickc Have you run any MNTK tests under `net45` with these changes?

**Comment by heynickc** [comment]
@alexvaluyskiy I just did and they fail, so I'm setting this as [WIP]

**Comment by alexvaluyskiy** [comment]
@heynickc thats why I did not push my changes

File: src/core/Akka.Remote.TestKit/MsgEncoder.cs [hunk 1]
```diff
@@ -7,126 +7,154 @@
...
+                var addressReply = (AddressReply)message;
+                wrapper.Addr = new Proto.Msg.AddressRequest
+                {
+                    Node = addressReply.Node.Name,
+                    Addr = AddressMessageBuilder(addressReply.Addr)
+                };
+            }
+            else if (message is Done)
+            {
+                wrapper.Done = " ";
```
Comment (heynickc) [inline src/core/Akka.Remote.TestKit/MsgEncoder.cs]: Protobuf encoder didn't like an empty string, didn't give it a type `{ done: ""}`, but instead just an empty string `{{ }}`.  Needs at least length > 0, but open to suggestions - tried String.Empty (doesn't work) and "done" which results in `{ "done": "done"}`

**Comment by heynickc** [comment]
MNTR specs passing now

File: src/core/Akka.Remote.TestKit/MsgEncoder.cs (hunk 1 above)
Comment (Aaronontheweb) [inline src/core/Akka.Remote.TestKit/MsgEncoder.cs]: I think that's fine

File: src/core/Akka.Remote.TestKit/Proto/ProtobufDecoder.cs [hunk 2]
```diff
@@ -5,49 +5,77 @@
...
-            var readable = input.ReadableBytes;
-            var buf = new byte[readable];
-            input.ReadBytes(buf);
-            var bs = ByteString.CopyFrom(buf);
-            var result = _extensions == null
-                ? _prototype.WeakCreateBuilderForType().WeakMergeFrom(bs).WeakBuild()
-                : _prototype.WeakCreateBuilderForType().WeakMergeFrom(bs, _extensions).WeakBuild();
-            output.Add(result);
+                //
+                // Note that we do not dispose the input stream because there is no input stream attached.
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote.TestKit/Proto/ProtobufDecoder.cs]: Helpful, thank you

**Pull request merged**